src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1427,14 +1450,13 @@
   (let [incoming-crls (utils/pem->crls incoming-crl-pem)
         current-crls (utils/pem->crls crl-path)
         cert-chain (utils/pem->certs cert-chain-path)
-        crls-by-issuer (->> incoming-crls
-                            (concat current-crls)
-                            set
-                            (group-by .getIssuerX500Principal))
-        new-crl-by-issuer (into {} (for [[issuer crls] crls-by-issuer]
-                                     [issuer (get-newest-crl crls)]))
-        new-crl-chain (map ((.getSubjectX500Principal ^X509Certificate %) new-crl-by-issuer)
-                           cert-chain)]
+        incoming-crls-by-issuer (->> incoming-crls
+                                     ;; even without combining with the current-crls,
+                                     ;; it doesn't seem like a bad idea to call `set`
+                                     set
```

Comment:
oh, I thought @justinstoller said he tried it and got identical objects? but I'm happy to remove this if it's not necessary.

Reply:
Tried this out:
```
dev-tools=> (count (set [(utils/pem->crl crl-path) (utils/pem->crl crl-path)]))
1
```
so I think it makes sense to keep `set`, just in case we get sent multiples of the same CRL (since it sounds like we're trying to be as permissive as is reasonably safe here).

Reply:
Thanks for trying this out! Since it works like you expected, it makes sense to keep it here. Sorry for the distraction :)